code/drasil-code-base/Language/Drasil/Code/Expr.hs

```diff
@@ -0,0 +1,135 @@
+{-# LANGUAGE GADTs #-}
+
+module Language.Drasil.Code.Expr where
+
+import Language.Drasil (UID, DomainDesc, Completeness, DerivType, RealInterval)
+
+-- Binary functions
+
+-- | Arithmetic operators (fractional, power, and subtraction).
+data ArithBinOp = Frac | Pow | Subt
```

Comment:
Maybe it would make sense to share all these `Op` types? They are not recursive, so they are independent.

Reply:
Would we ever want to add some sort of typing to `CodeExpr`? If we do, we might want to keep it as is and continue to have it closely follow `Expr` in development, otherwise, I think we should definitely share the `Op` types, it would definitely tidy things up!

Reply:
We will want to have both `Expr` and `CodeExpr` be typed. So if sharing might interfere with doing that, let's wait.

Reply:
Sounds good, let's wait a bit to see how it ends up. Thanks!